### PR TITLE
Removed condition to display thumb

### PR DESCRIPTION
**Pull request by mikayilsrt**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Condition removed for fix generateCategoriesMenu function to display categories thumbnails.
| Type?         | bug fix
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#20219.
| How to test?  | Install module and add new menu thumbnails from BackOffice.

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

**Comment by sarahdib** [comment]
Hello @mikayilsrt 

Could you target the dev branch  ?

Thank you

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by sarahdib** [comment]
Hello @mikayilsrt 
When I add a category in BO there is an error in FO : 
Capture d’écran 2020-10-26 à 19 52 05

File: ps_mainmenu.php [hunk 1]
```diff
@@ -743,19 +743,18 @@ protected function generateCategoriesMenu($categories, $is_children = 0)
             $node['label']   = $category['name'];
             $node['image_urls']  = [];
 
-            if (isset($category['children']) && !empty($category['children'])) {
+            if (isset($category['children']) && !empty($category['children']))
                 $node['children'] = $this->generateCategoriesMenu($category['children'], 1);
```
Comment (Progi1984) [inline ps_mainmenu.php:747]:
```suggestion
            if (isset($category['children']) && !empty($category['children'])) {
                $node['children'] = $this->generateCategoriesMenu($category['children'], 1);
            }
```

**Review by Progi1984 — COMMENTED** [review]
A small comment on coding style and it's OK for me.

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by sarahdib** [comment]
Capture d’écran 2020-11-05 à 10 29 46
Capture d’écran 2020-11-05 à 10 30 22
Thumb is not display in the menu

**Comment by mikayilsrt** [comment]
@sarahdib Hello,

I think your theme is overloading the .tpl file of the module, try to remove the module from the module folder of your theme in question.

**Comment by sarahdib** [comment]
@PrestaShop/prestashop-core-developers  can you check this in order to avoid the problem we had with the release of ps_searchbar ?  I can't validate this PR if I need to delete some file from the core.

**Comment by Progi1984** [comment]
@mikayilsrt Have you checked your PR with the classic theme ?

**Review by Progi1984 — APPROVED** [review]

File: ps_mainmenu.tpl [hunk 2]
```diff
@@ -22,30 +22,47 @@
  * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
  * International Registered Trademark & Property of PrestaShop SA
  *}
-{function name="menu" nodes=[] depth=0}
-  {strip}
+{assign var=_counter value=0}
+{function name="menu" nodes=[] depth=0 parent=null}
     {if $nodes|count}
-      <ul data-depth="{$depth}">
+      <ul class="top-menu" {if $depth == 0}id="top-menu"{/if} data-depth="{$depth}">
```
Comment (atomiix) [inline ps_mainmenu.tpl:28]:
```suggestion
      <ul class="top-menu" {if $depth == 0}id="top-menu"{/if} data-depth="{$depth}">
```
```suggestion
      <ul class="top-menu"{if $depth == 0} id="top-menu"{/if} data-depth="{$depth}">
```

File: ps_mainmenu.tpl [hunk 3]
```diff
@@ -22,30 +22,47 @@
...
+                {if $node.image_urls|count}
                   {foreach from=$node.image_urls item=image_url}
                     <img src="{$image_url}">
                   {/foreach}
-                </div>
+                {/if}
+                {$node.label}
+              </a>
+              {if $node.children|count}
+              <div {if $depth === 0} class="popover sub-menu js-sub-menu collapse"{else} class="collapse"{/if} id="top_sub_menu_{$_expand_id}">
```
Comment (atomiix) [inline ps_mainmenu.tpl:55]:
```suggestion
              <div {if $depth === 0}class="popover sub-menu js-sub-menu collapse"{else}class="collapse"{/if} id="top_sub_menu_{$_expand_id}">
```

**Review by atomiix — COMMENTED** [review]
Some cosmetic comments, otherwise the PR looks good 👍

**Review by atomiix — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by SD1982** [comment]
Hi @mikayilsrt I have the same results than @sarahdib 
No thumbnail in the menu
image

**Comment by mikayilsrt** [comment]
Hi @SD1982, Have you checked if your theme overloads the tpl file of the ps_mainmenu module? if this is the case delete the ps_mainmenu folder from your theme.

**Comment by SD1982** [comment]
Hi @mikayilsrt I use the classic theme provided by default with prestashop 

> Hi @SD1982, Have you checked if your theme overloads the tpl file of the ps_mainmenu module? if this is the case delete the ps_mainmenu folder from your theme.
@Progi1984 WDYT ?

**Comment by mikayilsrt** [comment]
@SD1982 Hi, in your classic theme go to open ps_mainmenu.tpl from module/ps_mainmenu folder and replace your code with https://github.com/mikayilsrt/ps_mainmenu/blob/master/ps_mainmenu.tpl

**Comment by SimonGrn** [comment]
> @SD1982 Hi, in your classic theme go to open ps_mainmenu.tpl from module/ps_mainmenu folder and replace your code with https://github.com/mikayilsrt/ps_mainmenu/blob/master/ps_mainmenu.tpl

I don't understand why we would need to do that. We are already replacing the code in `modules`ps_mainmenu` by the code in this PR, so it's your changes.

**Comment by PierreRambaud** [comment]
> > @SD1982 Hi, in your classic theme go to open ps_mainmenu.tpl from module/ps_mainmenu folder and replace your code with https://github.com/mikayilsrt/ps_mainmenu/blob/master/ps_mainmenu.tpl
> 
> I don't understand why we would need to do that. We are already replacing the code in `modules`ps_mainmenu` by the code in this PR, so it's your changes.

It's because themes override modules templates. In the classic theme, a lot of templates have been overridden :(

**Comment by khouloudbelguith** [comment]
Hi @mikayilsrt,

It is ok :heavy_check_mark: 
I tried with Classic theme & Different theme on PS1172 => OK
image
image

I tried with develop, PS1769 & 177x branch using Classic theme => ok
image


PS: I found an issue not related to this PR when editing a category using menu thumbs in PS1769 & PS1772 but it is fixed n the develop branch
https://drive.google.com/file/d/1RV52DEhjzrCByFe7_I1rIP2GjEruW0qw/view?usp=sharing

Thank you!

**Comment by Progi1984** [comment]
Thanks @mikayilsrt

**Pull request merged**